src/client/client.c

```diff
@@ -289,6 +294,171 @@ int method_lifecycle_action_on(Client *client, char *node_name, char *unit, char
         return r;
 }
 
+int method_metrics_toggle(Client *client, char *method) {
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        int r = 0;
+        char *result = NULL;
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        HIRTE_INTERFACE_BASE_NAME,
+                        HIRTE_OBJECT_PATH,
+                        MANAGER_INTERFACE,
+                        method,
+                        &error,
+                        &message,
+                        "");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        r = sd_bus_message_read(message, "s", &result);
+        if (r < 0) {
+                fprintf(stderr, "Failed to parse response message: %s\n", strerror(-r));
+                return r;
+        }
+        printf("%s\n", result);
+        return r;
+}
+
+static int match_start_unit_job_metrics_signal(
+                sd_bus_message *m, UNUSED void *userdata, UNUSED sd_bus_error *error) {
+        const char *job_path = NULL;
+        const uint64_t job_measured_time = 0;
+        const uint64_t unit_net_start_time = 0;
+        const int64_t hirte_time = 0;
+        const char *node_name = NULL;
+        int r = 0;
+
+        r = sd_bus_message_read(
+                        m, "ssttx", &node_name, &job_path, &job_measured_time, &unit_net_start_time, &hirte_time);
+        if (r < 0) {
+                fprintf(stderr, "Can't parse job result: %s", strerror(-r));
+                return r;
+        }
+
+        printf("[%s] Start job %s:\n\t"
+               "Hirte job gross measured time: %ldms\n\t"
+               "Unit net start time (from properties): %ldms,\n\t"
```

Comment:
When testing manually a bit, the `unit net start time` has always been 0. In addition, I get a generic numeric time for systemd printed - which is the net time that should be displayed here, right? For example: 
```bash
Generic numeric [pi] Agent systemd job net measured time: 145
[pi] Start job /org/containers/hirte/job/3:
        Hirte job gross measured time: 154ms
        Unit net start time (from properties): 0ms,
        Hirte time: 154ms
 ```
 What is the reason behind the generic numeric/string signals? Can't we collect it all in one signal instead of spreading out to different ones?

Reply:
I have removed the generic metrics and created a specific one for agent job.